uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/manage-leave-requests.spec.js

```diff
@@ -441,11 +448,19 @@
                 controller.refresh();
               });
 
-              it('Contact API gets called with region value', function () {
-                expect(Contact.leaveManagees).toHaveBeenCalledWith(controller.contactId, jasmine.objectContaining({
-                  region: mockRegion
-                }));
-              });
+              if (isAdmin) {
```

Comment:
1. In the current setup I dont think it is even testing both the scenarios, because you are not changing the value of isAdmin.
2. Having if-else blocks in the unit test should be avoided.
3. Multiple test cases must be test up to test different scenarios
```javascript
describe('when admin', function() {
 beforeEach(function(){
   // setAdmin();
 })
  // Test every case where admin needs to be true in this describe
})
describe('when not admin', function() {
 beforeEach(function(){
   // setManager or staff();
 })
  // Test every case where admin needs to be false in this describe
})
```
Please check all such scenarios